packages/@aws-cdk/aws-scheduler-alpha/lib/target.ts

```diff
@@ -0,0 +1,231 @@
+   *
+   * Minimum value of 60.
+   * Maximum value of 86400.
+   *
+   * @default Duration.hours(24)
+   */
+  readonly maxEventAge?: Duration;
+  /**
+   * The maximum number of times to retry when the target returns an error.
+   *
+   * Minimum value of 0.
+   * Maximum value of 185.
+   *
+   * @default 185
+   */
+  readonly retryAttempts?: number;
+}
+
+/**
+ * Properties for an Schedule Target
+ */
+export interface ScheduleTargetConfig {
+  /**
+   * The Amazon Resource Name (ARN) of the target.
+   */
+  readonly arn: string;
+
+  /**
+   * Role to use to invoke this event target
+   */
+  readonly role: iam.IRole;
+
+  /**
+   *  What input to pass to the tatget
+   */
+  readonly input?: ScheduleTargetInput;
+
+  /**
+   * A `RetryPolicy` object that includes information about the retry policy settings, including the maximum age of an event, and the maximum number of times EventBridge Scheduler will try to deliver the event to a target.
+   */
+  readonly retryPolicy?: CfnSchedule.RetryPolicyProperty;
+
+  /**
+   * An object that contains information about an Amazon SQS queue that EventBridge Scheduler uses as a dead-letter queue for your schedule. If specified, EventBridge Scheduler delivers failed events that could not be successfully delivered to a target to the queue.\
+   */
+  readonly deadLetterConfig?: CfnSchedule.DeadLetterConfigProperty
+
+  /**
+   *  The templated target type for the Amazon ECS RunTask API Operation.
+   */
+  readonly ecsParameters?: CfnSchedule.EcsParametersProperty;
+  /**
+   * The templated target type for the EventBridge PutEvents API operation.
+   */
+  readonly eventBridgeParameters?: CfnSchedule.EventBridgeParametersProperty;
+
+  /**
+   * The templated target type for the Amazon Kinesis PutRecord API operation.
+   */
+  readonly kinesisParameters?: CfnSchedule.KinesisParametersProperty;
+
+  /**
+   * The templated target type for the Amazon SageMaker StartPipelineExecution API operation.
+   */
+  readonly sageMakerPipelineParameters?: CfnSchedule.SageMakerPipelineParametersProperty;
+  /**
+   * The templated target type for the Amazon SQS SendMessage API Operation
+   */
+  readonly sqsParameters?: CfnSchedule.SqsParametersProperty;
+}
+
+/**
+ * Bind props to base schedule target config.
+ * @internal
+ */
+export function bindBaseTargetConfig(props: ScheduleTargetBaseProps) {
+  let { deadLetterQueue } = props;
+
+  return {
+    deadLetterConfig: deadLetterQueue ? { arn: deadLetterQueue?.queueArn } : undefined,
+    retryPolicy: renderRetryPolicy(props.maxEventAge, props.retryAttempts),
+  };
+}
+
+/**
+ * Obtain the Role for the EventBridge Schedule
+ *
+ * If a role already exists, it will be returned. This ensures that if multiple
+ * events have the same target, they will share a role.
+ * @internal
+ */
+export function singletonEventRole(scope: IConstruct): iam.IRole {
+  const id = 'SchedulesRole';
```

Comment:
In the code we prepared this had a lot more beef.

1. You use one role named 'SchedulesRole' for all targets in the current stack. The more targets there are the less "least privilege' this will get. Personally I think one role per target makes more sense then one role per stack.
2. The code we started with had Stack.of(schedule) and then stack.node.tryFindChild(id) in stead of scope.node.tryFindChild. If you find the role for a target it is fine.

Reply:
3. Why do we not also narrow down of the use of the role to "aws:SourceArn" "aws:SourceAccount"

Reply:
I originally preferred this implementation inspired by `events.target` (https://github.com/aws/aws-cdk/blob/e479bd4353aefa5e48189d2c71f6067489afe141/packages/aws-cdk-lib/aws-events-targets/lib/util.ts#L63-L81) as it seemed simpler to me. However, I agree that the least privilege policy is a good point. Now I understand where the complexity is coming from.